.github/workflows/build-kata-static-tarball-s390x.yaml

```diff
@@ -75,6 +85,25 @@ jobs:
           TARGET_BRANCH: ${{ inputs.target-branch }}
           RELEASE: ${{ inputs.stage == 'release' && 'yes' || 'no' }}
 
+      - uses: oras-project/setup-oras@v1
+        if: (matrix.asset == 'agent') && (inputs.push-to-registry == 'yes')
+
+      # for pushing attestations to the registry
+      - uses: docker/login-action@v3
+        if: (matrix.asset == 'agent') && (inputs.push-to-registry == 'yes')
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: actions/attest-build-provenance@v1
+        if: (matrix.asset == 'agent') && (inputs.push-to-registry == 'yes')
+        with:
+          subject-name: ${{ steps.build.outputs.oci-name }}
+          subject-digest: ${{ steps.build.outputs.oci-digest }}
+          push-to-registry: true
```

Comment:
So just to double check I understand - we build the provenance here and that will them be available and linked when we push the cache in kata-deploy-binaries. Is there anything special we need to do to make that linking work?

Reply:
no that's it. the linking happens via the [referrer](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md#manifest-referrers-api) api in manifests/registries. you can find (e.g. via oras discover) referrer of a given manifest.

to retrieve the attestation you don't need to access the github api, but can start from an oci image and discover the attestation for it.

Reply:
Sorry, just to update here, I'd missed that `${build_target}-oci-image` was created by the oras push and then used in the build provenance, so that was why I was confused by the magical linking :)